custom_linting_rules/noExternalHttpLinkRule.js

```diff
@@ -0,0 +1,100 @@
+        type: 'functionality',
+        description: 'Anchor tags with an external link must use https',
+        options: null,
+        optionsDescription: '',
+        typescriptOnly: true,
+        issueClass: 'SDL',
+        issueType: 'Error',
+        severity: 'Critical',
+        level: 'Mandatory',
+        group: 'Security',
+        commonWeaknessEnumeration: '242,676'
+    };
+    return Rule;
+}(Lint.Rules.AbstractRule));
+exports.Rule = Rule;
+var NoExternalHttpLinkRuleWalker = /** @class */ (function (_super) {
+    __extends(NoExternalHttpLinkRuleWalker, _super);
+    function NoExternalHttpLinkRuleWalker() {
+        return _super !== null && _super.apply(this, arguments) || this;
+    }
+    NoExternalHttpLinkRuleWalker.prototype.visitJsxElement = function (node) {
+        var openingElement = node.openingElement;
+        this.validateOpeningElement(openingElement);
+        _super.prototype.visitJsxElement.call(this, node);
+    };
+    NoExternalHttpLinkRuleWalker.prototype.visitJsxSelfClosingElement = function (node) {
+        this.validateOpeningElement(node);
+        _super.prototype.visitJsxSelfClosingElement.call(this, node);
+    };
+    NoExternalHttpLinkRuleWalker.prototype.validateOpeningElement = function (openingElement) {
+        if (openingElement.tagName.getText() === 'a') {
+            var allAttributes = JsxAttribute_1.getJsxAttributesFromJsxElement(openingElement);
+            /* tslint:disable:no-string-literal */
+            var href = allAttributes['href'];
+            /* tslint:enable:no-string-literal */
+            // if (target != null && getStringLiteral(target) === '_blank' && !isRelAttributeValue(rel)) {
+            //     this.addFailureAt(openingElement.getStart(), openingElement.getWidth(), FAILURE_STRING);
+            // }
+            if (href != null && !isSafeHrefAttributeValue(href)) {
+                var fix = new Lint.Replacement(openingElement.getStart(), openingElement.getWidth(), "");
+                this.addFailureAt(openingElement.getStart(), openingElement.getWidth(), FAILURE_STRING, fix);
+            }
+        }
+    };
+    return NoExternalHttpLinkRuleWalker;
+}(ErrorTolerantWalker_1.ErrorTolerantWalker));
+function isSafeHrefAttributeValue(attribute) {
+    if (JsxAttribute_1.isEmpty(attribute)) {
+        return false;
+    }
+    if (attribute.initializer.kind === ts.SyntaxKind.JsxExpression) {
+        var expression = attribute.initializer;
+        if (expression.expression != null && expression.expression.kind !== ts.SyntaxKind.StringLiteral) {
+            return true; // attribute value is not a string literal, so do not validate
+        }
+    }
+    var stringValue = JsxAttribute_1.getStringLiteral(attribute);
+    if (stringValue === '#') {
+        return true;
+    }
+    else if (stringValue == null || stringValue.length === 0) {
+        return false;
+    }
+    return stringValue.indexOf('https://') >= 0;
+}
```

Comment:
It would be ideal if we didn't check in compiled code, but I'm not sure what the linting process would look like in that case. I think this is OK for now.

Reply:
I agree. According to the docs found [here](https://palantir.github.io/tslint/develop/custom-rules/) (final notes section at the end), we can run the rule in typescript. However, I tried running this way and couldn't get it to work